src/Data/ContentSearchResult/ContentSearchResultDataClass.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  currentLanguage,
+  extractText,
+  load,
+  Obj,
+  provideDataClass,
+  urlFor,
+} from 'scrivito'
+import { objTitle } from '../../utils/objTitle'
+import { ensureString } from '../../utils/ensureString'
+import { isImage } from '../../Objs/Image/ImageObjClass'
+
+const BLACKLIST_OBJ_CLASSES = ['Image', 'Redirect', 'Video']
+
+export const ContentSearchResult = provideDataClass('ContentSearchResult', {
+  attributes: async () => {
+    const lang = await load(currentLanguage)
+
+    return {
+      image: [
+        'reference',
+        { title: lang === 'de' ? 'Bild' : 'Image', to: 'Image' },
+      ],
+      snippet: ['string', { title: lang === 'de' ? 'Schnipsel' : 'Snippet' }],
+      title: ['string', { title: lang === 'de' ? 'Titel' : 'Title' }],
+      url: ['string', { title: 'URL' }],
+    }
+  },
+  title: async () =>
+    (await load(currentLanguage)) === 'de'
+      ? 'Inhalts-Suchergebnis'
+      : 'Content Search Result',
+  connection: {
+    async index(params) {
+      if (Object.keys(params.filters()).length > 0) {
+        throw new Error('Filtering is not supported for CMS Search Results.')
+      }
+
+      if (params.order().length > 0) {
+        throw new Error('Sorting is not supported for CMS Search Results.')
+      }
+
+      const search = params.search()
+      if (!search) return { results: [], count: 0 }
+
+      const objSearch = Obj.whereFullTextOf('*', 'matches', search)
+        .andNot('_objClass', 'equals', BLACKLIST_OBJ_CLASSES)
+        .and('_dataParam', 'equals', null) // Ignore data details pages
```

Comment:
* [ ] Undocumented feature. Is there an issue for that? Maybe add a bit more about how/why/for how long this works to the comment? How about removing this magic and setting `excludeFromSearch: true` for these pages?

Reply:
I don't think that there is an issue for that. It is also not mentioned in the original introduction in https://github.com/Scrivito/scrivito-portal-app/commit/0f854f830452aab04c738753e588e90c595f75f2.

Reply:
I'm hesitant to manually set `excludeFromSearch: true` for detail pages. This could be a trap, since the editor would have to think about this whenever he or she creates a new details page. I think it's better to filter them out by default.

Reply:
I created a new issue of this: https://github.com/infopark/scrivito_js/issues/11592